addons/binding/org.openhab.binding.evohome/src/main/java/org/openhab/binding/evohome/handler/EvohomeAccountBridgeHandler.java

```diff
@@ -0,0 +1,263 @@
+import org.openhab.binding.evohome.internal.api.models.v2.response.GatewayStatus;
+import org.openhab.binding.evohome.internal.api.models.v2.response.Location;
+import org.openhab.binding.evohome.internal.api.models.v2.response.LocationStatus;
+import org.openhab.binding.evohome.internal.api.models.v2.response.Locations;
+import org.openhab.binding.evohome.internal.api.models.v2.response.LocationsStatus;
+import org.openhab.binding.evohome.internal.api.models.v2.response.TemperatureControlSystem;
+import org.openhab.binding.evohome.internal.api.models.v2.response.TemperatureControlSystemStatus;
+import org.openhab.binding.evohome.internal.api.models.v2.response.Zone;
+import org.openhab.binding.evohome.internal.api.models.v2.response.ZoneStatus;
+import org.openhab.binding.evohome.internal.configuration.EvohomeAccountConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Provides the bridge for this binding. Controls the authentication sequence.
+ * Manages the scheduler for getting updates from the API and updates the Things it contains.
+ *
+ * @author Jasper van Zuijlen - Initial contribution
+ *
+ */
+public class EvohomeAccountBridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(EvohomeAccountBridgeHandler.class);
+    private EvohomeAccountConfiguration configuration;
+    private EvohomeApiClient apiClient;
+    private List<AccountStatusListener> listeners = new CopyOnWriteArrayList<AccountStatusListener>();
+
+    protected ScheduledFuture<?> refreshTask;
+
+    public EvohomeAccountBridgeHandler(Bridge thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        configuration = getConfigAs(EvohomeAccountConfiguration.class);
+
+        if (checkConfig()) {
+            try {
+                apiClient = new EvohomeApiClientV2(configuration);
+
+                // Initialization can take a while, so kick if off on a separate thread
+                scheduler.schedule(new Runnable() {
+                    @Override
+                    public void run() {
+                        if (apiClient.login()) {
+                            if (checkInstallationInfoHasDuplicateIds(apiClient.getInstallationInfo())) {
+                                startRefreshTask();
+                            } else {
+                                updateAccountStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                        "System Information Sanity Check failed");
+                            }
+                        } else {
+                            updateAccountStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                    "Authentication failed");
+                        }
+                    }
+                }, 0, TimeUnit.SECONDS);
+            } catch (Exception e) {
```

Comment:
This catch should be around the `apiClient` line only.

Reply:
Not sure what you mean by that? I don't want to schedule the task when the EvohomeApiClientV2 throws and in `run` the apiClient is used on line 77 & 78?